### PR TITLE
Added dind-golang docker image

### PR DESCRIPTION
**Pull request by neel-astro**
## Which issue this PR fixes

https://github.com/astronomer/issues/issues/3818

## Summary of changes

Added dind-golang image to ap-vendor with golang version 1.17

## Images are updated by this PR

- [ ] alertmanager
- [ ] blackbox-exporter
- [ ] configmap-reloader
- [ ] curator
- [ ] elasticsearch
- [ ] elasticsearch-exporter
- [ ] fluentd
- [ ] git-sync
- [ ] grafana
- [ ] keda
- [ ] keda-metrics-apiserver
- [ ] kibana
- [ ] kube-state
- [ ] kubed
- [ ] nats-server
- [ ] nats-streaming
- [ ] nats-exporter
- [ ] nginx
- [ ] nginx-es
- [ ] node-exporter
- [ ] openresty
- [ ] pgbouncer
- [ ] pgbouncer-exporter
- [ ] postgresql
- [ ] postgres-exporter
- [ ] prometheus
- [ ] redis
- [ ] registry
- [ ] statsd-exporter

## Checklist

Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.

- [ ] version.txt is updated in the changed image(s)

If a security scan fails for an unchanged image...

- [ ] a fix has been applied OR...
- [ ] an [issue]() has been created

If adding a new image:

- [x] the directory has the same name you intend it to be published as, less a preceding "ap-"
- [x] the directory includes a file "version.txt", with version matching the underlying software version
- [x] the file .github/PULL_REQUEST_TEMPLATE.md is updated to include the image in the checklist
- [x] execute the script .circleci/generate_circleci_config.py, commit changes to .circleci/config.yml

### CODE REVIEW

File: dind-golang/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        export PATH="/usr/local/go/bin:$PATH"; \
+        go version
+
+ENV GOPATH /go
+ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+
+RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.37.1
+
+RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
+WORKDIR $GOPATH
```
Comment (andriisoldatenko) [inline dind-golang/Dockerfile:69]:
```suggestion
WORKDIR $GOPATH

```

**Comment by danielhoherd** [comment]
Will need to create repos on quay and docker hub before merging this.

**Comment by neel-astro** [comment]
> Will need to create repos on quay and docker hub before merging this.

Who is currently maintaining quay and docker hub, and is authorized to create new repos?
cc @danielhoherd

**Comment by danielhoherd** [comment]
I created the repos on quay and docker hub.

**Comment by danielhoherd** [comment]
Verified that the push of this new image worked. <https://app.circleci.com/pipelines/github/astronomer/ap-vendor/704/workflows/4193eb24-e22a-451c-b477-f5173ae3a3ec/jobs/32570>

**Pull request merged**